src/firefly/js/tables/TableUtil.js

```diff
@@ -228,6 +233,41 @@ export function smartMerge(target, source) {
     }
 }
 
+/**
+ * sort the given tableModel based on the given request
+ * @param origTableModel original table model.  this is returned when direction is UNSORTED.
+ * @param sortInfoStr
+ */
+export function sortTable(origTableModel, sortInfoStr) {
+    const sortInfoCls = SortInfo.parse(sortInfoStr);
+    const colName = get(sortInfoCls, 'sortColumns.0');
+    const dir = get(sortInfoCls, 'direction', UNSORTED);
+    if (dir === UNSORTED || get(origTableModel, 'tableData.data.length', 0) === 0) return origTableModel;
+
+    const multiplier = dir === SORT_ASC ? 1 : -1;
+    const colIdx = findColumnIdx(origTableModel, colName);
+    const col = get(origTableModel, ['tableData','columns', colIdx]);
+
+    var tableModel = cloneDeep(origTableModel);
+    set(tableModel, 'request.sortInfo', sortInfoStr);
+
+    var comparator;
+    if (!col.type || ['char', 'c'].includes(col.type) ) {
+        comparator = (r1, r2) => {
+                const [s1, s2] = [r1[colIdx], r2[colIdx]];
+                return multiplier * (s1 > s2 ? 1 : -1);
+            };
+    } else {
+        comparator = (r1, r2) => {
+                const [v1, v2] = [r1[colIdx], r2[colIdx]];
+                return multiplier * (Number(v1) - Number(v2));
+            };
```

Comment:
it will still work okay since comparator only check for greater/less than zero.  i've tried it, and it did not crash.